lms/djangoapps/student_account/views.py

```diff
@@ -107,6 +110,11 @@ def login_and_registration_form(request, initial_mode="login"):
     if ext_auth_response is not None:
         return ext_auth_response
 
+    rendered_enterprise_sidebar = enterprise_sidebar(request)
+
+    if rendered_enterprise_sidebar:
+        form_descriptions['registration'] = enterprise_fields_only(form_descriptions['registration'])
+
```

Comment:
To make this less intrusive, could we do something like this instead?

```python
    if ext_auth_response is not None:
          return ext_auth_response
    context = {
          'data': {
          ...
           'disable_footer': not configuration_helpers.get_value(
            'ENABLE_COMBINED_LOGIN_REGISTRATION_FOOTER',
            settings.FEATURES['ENABLE_COMBINED_LOGIN_REGISTRATION_FOOTER']
        ),
    }

    context = enterprise_update_context(request, context)
```

and move the sidebar and registration field manipulations to a new method, `enterprise_update_context()`?

Then it's a one-line change to this view, and you don't have to unpack and re-pack the JSON for the registration fields.